blockprotocol/blockprotocol
# Add "Tax ID" field to billing page

## 🌟 What is the purpose of this PR?




This PR adds a "Tax ID" section to the billing page for users with a stripe subscription.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203623179643287/1203880493107350/f) _(internal)_

## 🔍 What does this change?




See commits.

## 📜 Does this require a change to the docs?



Not that I'm aware of.

## ⚠️ Known issues



## 🛡 What tests cover this?



Manual testing.

## ❓ How to test this?



1. Checkout the branch
2. Add the `NEXT_PUBLIC_BILLING_FEATURE_FLAG=1` env var to your `app/site/.env.local` to enable the billing feature flag
3. Run the local developer environment for the BP site
4. Checkout the HASH internal API repo on the `bw/set-tax-id` branch (if it hasn't merged yet)
5. Follow the instructions in the internal API README to run the local developer environment, except for the mongo db external service (as the BP site is already running mongo db)
6. Login as a user and navigate to `/settings/billing`
7. Ensure the logged in user is subscribed to a paid subscription tier (by upgrading to "hobby" or "pro" for example)
8. Observe the "Tax ID" input and ensure you're able to input a valid Tax ID for a region

## 📹 Demo



https://user-images.githubusercontent.com/42802102/221115934-db7a26bb-817b-4012-93ba-1e64e2353730.mov
